sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -659,604 +600,482 @@ protected EventProcessorClient()
+                {
+                    FullyQualifiedNamespace = FullyQualifiedNamespace,
+                    EventHubName = EventHubName,
+                    ConsumerGroup = ConsumerGroup,
+                    PartitionId = context.PartitionId,
+                    StartingPosition = EventPosition.FromOffset(eventData.Offset)
+                };
 
-            try
-            {
-                return StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken);
+                using var scope = EventDataInstrumentation.ScopeFactory.CreateScope(DiagnosticProperty.EventProcessorCheckpointActivityName);
+                scope.Start();
+
+                try
+                {
+                    return StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken);
+                }
+                catch (Exception ex)
+                {
+                    scope.Failed(ex);
+                    throw;
+                }
             }
-            catch (Exception e)
+            catch (Exception ex)
             {
                 // In case of failure, there is no need to call the error handler because the exception can
-                // be thrown directly to the user here.
+                // be thrown directly to the caller here.
 
-                scope.Failed(e);
+                Logger.UpdateCheckpointError(context.PartitionId, Identifier, EventHubName, ConsumerGroup, ex.Message);
                 throw;
             }
             finally
             {
-                Logger.UpdateCheckpointComplete(context.PartitionId);
+                Logger.UpdateCheckpointComplete(context.PartitionId, Identifier, EventHubName, ConsumerGroup);
             }
         }
 
         /// <summary>
-        ///   Creates an <see cref="EventHubConsumerClient" /> to use for processing.
+        ///   Creates an <see cref="EventHubConnection" /> to use for communicating with the Event Hubs service.
         /// </summary>
         ///
-        /// <param name="consumerGroup">The consumer group to associate with the consumer client.</param>
-        /// <param name="connection">The connection to use for the consumer client.</param>
-        /// <param name="options">The options to use for configuring the consumer client.</param>
-        ///
-        /// <returns>An <see cref="EventHubConsumerClient" /> with the requested configuration.</returns>
+        /// <returns>The requested <see cref="EventHubConnection" />.</returns>
         ///
-        internal virtual EventHubConsumerClient CreateConsumer(string consumerGroup,
-                                                               EventHubConnection connection,
-                                                               EventHubConsumerClientOptions options) => new EventHubConsumerClient(consumerGroup, connection, options);
+        protected override EventHubConnection CreateConnection()
+        {
+            if (ConnectionFactory != null)
```

Comment:
It's possible, but it is awkward due to the way that Moq exposes protected members; you lose strong typing and end up with magic strings.   It's not terrible, but it does make discoverability for those writing tests harder.  If we went down that route, in the raw it would look something like:

```csharp
var mockConnection = Mock.Of<EventHubConnection>();
var mockProcessor = new Mock<EventProcessorClient>( ... STUFF ...) { Callbase = true };

mockProcessor
    .Protected()
    .Setup<EventHubConnection>("CreateConnection")
    .Returns(mockConnection);
```

Granted we could work around that with some test shims or whatnot.   Given that we ended up writing a local test class to wrap a bunch of private methods, it may be worth giving it a look and just using that to shim.    I'll kick the tires and see how much churn it causes.   I'm a bit short on runway between now and code complete, so I'll likely defer if it causes non-trivial rework.

Reply:
> I'll kick the tires and see how much churn it causes. I'm a bit short on runway between now and code complete, so I'll likely defer if it causes non-trivial rework.

Sounds good.  Feel free to defer and thanks for the detailed explanation.

Reply:
This seems to have worked out with minimal effort; the change is included.